src/Npgsql/NpgsqlBinaryExporter.cs

```diff
@@ -127,7 +127,48 @@ async Task ReadHeader(bool async)
     /// The number of columns in the row. -1 if there are no further rows.
     /// Note: This will currently be the same value for all rows, but this may change in the future.
     /// </returns>
-    public int StartRow() => StartRow(false).GetAwaiter().GetResult();
+    public int StartRow()
```

Comment:
This is a per-row thing, so a bit less convincing than per-column in terms of needing to split sync/async for micro-optimization, no?

Reply:
Essentially, as you can see in the results, this is anything but a micro-optimization. It's really just quite expensive to run an async method for a small bit of work. Additionally it thwarts all inlining from the JIT due to the try finally present in AsyncMethodBuilder.Start (which is i.a. responsible for getting and resetting the sync ctx, thread static ops that are also not free)

I agree the frequency is lower but IMO it's still important, we try hard in the reader to stay sync as well.